src/esp/gfx/RenderCamera.cpp

```diff
@@ -155,5 +155,30 @@ uint32_t RenderCamera::draw(MagnumDrawableGroup& drawables, Flags flags) {
   return drawableTransforms.size();
 }
 
+esp::geo::Ray RenderCamera::unproject(const Mn::Vector2i& viewportPosition) {
+  esp::geo::Ray ray;
+  ray.origin = object().absoluteTranslation();
+  if (viewportPosition.min() < 0 || (viewport() - viewportPosition).min() < 0) {
+    LOG(WARNING) << "RenderCamera::unproject : viewportPosition outside of "
```

Comment:
Is this necessary? I guess a screen-space point outside the viewport can still be unprojected using your math below.

An example use case would be taking the cursor position and then padding it in all directions by some screen-space margin to produce a "cone" of rays, with part of the cone extending outside the viewport.